### PR TITLE
[ADD][account_fiscalyear_close]Module

### PR DESCRIPTION
**Pull request by lmgr0312**
@ougc27

### CODE REVIEW

File: account_fiscalyear_close/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2018 Jarsa Sistemas, S.A. de C.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Account Fiscal Year Close",
+    "summary": "Account Fiscal Year Close",
+    "version": "10.0.1.0.0",
+    "category": "Hidden",
```
Comment (ougc27) [inline account_fiscalyear_close/__manifest__.py]: Change for "Accounting & Finance'

File: account_fiscalyear_close/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "website": "https://www.jarsa.com.mx/",
+    "author": "Jarsa Sistemas",
+    "license": "AGPL-3",
+    "installable": True,
```
Comment (ougc27) [inline account_fiscalyear_close/__manifest__.py]: This key is not necessary, please remove it

File: account_fiscalyear_close/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "depends": [
+        'account_accountant',
+    ],
+    "data": [
+        'wizard/account_fiscalyear_close_wizard_view.xml',
+    ],
+    "demo": [
```
Comment (ougc27) [inline account_fiscalyear_close/__manifest__.py]: Remove this empty key

File: account_fiscalyear_close/wizard/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2018 Jarsa Sistemas, S.A. de C.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import account_fiscalyear_close_wizard
```
Comment (ougc27) [inline account_fiscalyear_close/wizard/__init__.py:5]: Add a extra line at the end of file

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string="Journal",
+        domain=[('type', '=', 'general')],
+        required=True,
+    )
+
+    @api.model
+    def get_date_start(self):
+        return str(fields.date.today().year - 1)+"-01-01"
```
Comment (ougc27) [inline account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: When you concatenate a string use spaces to separate each part of the concatenation Eg.

`return str(fields.date.today().year - 1) + "-01-01"`

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    @api.model
+    def get_date_end(self):
+        return str(fields.date.today().year - 1)+"-12-31"
```
Comment (ougc27) [inline account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: Same case of concatenation

**Comment by ougc27** [comment]
Nice MR

Please check Travis and fix the issues. :+1:

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018, Jarsa Sistemas, S.A. de C.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
```
Comment (ougc27) [inline account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: Remove this unused library.

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
```
Comment (ougc27) [inline account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: Remove this unused library.

**Review by ougc27 — CHANGES_REQUESTED** [review]

File: account_fiscalyear_close/__manifest__.py (hunk 1 above)
Comment (lmgr0312) [reply account_fiscalyear_close/__manifest__.py]: ok

File: account_fiscalyear_close/__manifest__.py (hunk 2 above)
Comment (lmgr0312) [reply account_fiscalyear_close/__manifest__.py]: ok

File: account_fiscalyear_close/__manifest__.py (hunk 3 above)
Comment (lmgr0312) [reply account_fiscalyear_close/__manifest__.py]: ok

File: account_fiscalyear_close/wizard/__init__.py (hunk 4 above)
Comment (lmgr0312) [reply account_fiscalyear_close/wizard/__init__.py:5]: ok

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py (hunk 5 above)
Comment (lmgr0312) [reply account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: ok

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py (hunk 6 above)
Comment (lmgr0312) [reply account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: ok

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py (hunk 7 above)
Comment (lmgr0312) [reply account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: ok

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py (hunk 8 above)
Comment (lmgr0312) [reply account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py]: ok, all the changes are done

**Review by ougc27 — APPROVED** [review]

**Comment by ougc27** [comment]
Hello @alan196 this PR is good for me, what do you think? Also, the runbot parameter is required but i think that is not working.

File: account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    def get_years_ending_entry(self):
+        self.ensure_one()
+        am_obj = self.env['account.move']
+        aml_obj = self.env['account.move.line']
+        move = am_obj.create(self.prepare_am())
+        data = self.get_account_balances()
+        for item in data:
+            root_aml = aml_obj.with_context(check_move_validity=False).create(
+                self.prepare_aml(item, move))
+            root_aml.copy({
```
Comment (alan196) [inline account_fiscalyear_close/wizard/account_fiscalyear_close_wizard.py:81]: why this copy?

you are creating a line for each account?

**Review by alan196 — COMMENTED** [review]
I need to understand the reason for the copy.

**Pull request merged**